src/Orleans.TestingHost/TestStorageProviders/FaultInjectionStorageProvider.cs

```diff
@@ -1,73 +1,52 @@
 ﻿
 using System;
+using System.Dynamic;
 using System.Threading.Tasks;
 using Microsoft.Extensions.Logging;
 using Orleans.Providers;
 using Orleans.Runtime;
 using Orleans.Storage;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
 
 namespace Orleans.TestingHost
 {
+    /// <summary>
+    /// Options for fault injection grain storage
+    /// </summary>
+    public class FaultInjectionGrainStorageOptions
+    {
+        public static TimeSpan DEFAULT_LATENCY = TimeSpan.FromMilliseconds(10);
+        /// <summary>
+        /// Latency applied on storage operation
+        /// </summary>
+        public TimeSpan Latency { get; set; } = DEFAULT_LATENCY;
+    }
     /// <summary>
     /// Fault injection decorator for storage providers.  This allows users to inject storage exceptions to test error handling scenarios.
     /// </summary>
     /// <typeparam name="TStorage"></typeparam>
-    public class FaultInjectionStorageProvider<TStorage> : IStorageProvider
-        where TStorage : IStorageProvider, new()
+    public class FaultInjectionGrainStorage : IGrainStorage, ILifecycleParticipant<ISiloLifecycle>
```

Comment:
we should refactor this pattern entirely, but out of scope for this PR.

Reply:
agree